### DISCUSSION TITLE
Solving expressions that have '||' or '&&' operators

### DISCUSSION THREAD

**Discussion by m7mdtiger** [josdejong/mathjs#2920] [Show and tell]
Hello, 
I have read that you prefer not to add solving expressions that have '||' or '&&' operators to the library.  Even though I think that would be pretty useful especially that the conditional operation is already added to the library.

Anyway I had to solve this issue by going around it. And I am sharing it as others might find this helpful..

First I use **mathjs.evaluate()** function. And if I still have unsolved expressions then I use eval() function from JS like this:
```
if (Object.keys(expressions).length > 0) {
    let counter = 20;
    while (Object.keys(expressions).length > 0 && counter > 0) {
      for (const key in expressions) {
        try {
          expressions[key] = expressions[key]
            .replace(/&&/g, "and")
            .replace(/\|\|/g, "or");
          const node = math.parse(expressions[key]);
          const transformed = node.transform(function (node) {
            if (
              node.isSymbolNode &&
              Object.prototype.hasOwnProperty.call(scope, node.name)
            ) {
              return new math.ConstantNode(scope[node.name]);
            } else {
              return node;
            }
          });

          const transformedExpression = transformed.toString();
          const replacedExpression = transformedExpression
            .replace(/and/g, "&&")
            .replace(/or/g, "||");
          const evaluatedValue = eval(replacedExpression);
          scope[key] = evaluatedValue;
          solvedExpressions[key] = evaluatedValue;
          delete expressions[key];
        } catch (error) {
          console.warn(
            `Undefinde expression trying to solve `,
            key,
            "error msg: ",
            error.message
          );
          counter--;
        }
      }
    }
  }
```

Thank you a lot for your efforts.

**Comment by josdejong** [comment]
Thanks for sharing @m7mdtiger. The expression parser of mathjs indeed has logical operators `and` and `or` instead of `&&` and `||`.

**Reply by m7mdtiger** [reply]
Yes, But I couldn't solve expressions having **and** or **or**  when trying to solve expressions with mathjs.evaluate() function and not using the parser function. 

Thank you ^^

**Reply by josdejong** [reply]
That makes me curious: why do you want to use the JavaScript `eval` over the expression parser it self? (or do I misunderstand you?)

**Reply by m7mdtiger** [reply]
Oh lovely, I've tried several times yesterday replacing **||** and **&&** by **or** and **and**. And then use math.evaluate() function but that didn't work. I have tried one more time just now and everything is good xD

Thank you so much @josdejong  :flushed:
P.S by the end of the day I got lazy to run my code and used a playground that I found on google :v Maybe I was using an older version at that point :v

**Reply by josdejong** [reply]
Good to hear👍 

Best to avoid the JavaScript `eval` function if possible, it can be unsafe to use.

**Reply by m7mdtiger** [reply]
Sure, thank you a lot ^^
